server/reload.go

```diff
@@ -41,6 +42,20 @@ func (d *debugOption) Apply(server *Server) {
 	server.Noticef("Reloaded: debug = %v", d.newValue)
 }
 
+// tlsOption implements the option interface for the `tls` setting.
+type tlsOption struct {
+	newValue *tls.Config
+}
+
+// Apply the tls change.
+func (t *tlsOption) Apply(server *Server) {
+	tlsRequired := t.newValue != nil
+	server.info.TLSRequired = tlsRequired
+	server.info.TLSVerify = tlsRequired && t.newValue.ClientAuth == tls.RequireAndVerifyClientCert
```

Comment:
We are not returning the value of an assignment. This is purely a boolean expression being assigned to `server.info.TLSVerify`. Maybe the parens used [here](https://github.com/nats-io/gnatsd/blob/master/server/server.go#L104) make that more clear?